app/helpers/selections_helper.rb

```diff
@@ -3,48 +3,39 @@
 
 module SelectionsHelper
 
-  def get_forecast(stations)
+  def get_forecast(selections)
 
-    w_api = Wunderground.new("dda1e42fba07eb1c")
-    current_user.forecast = [];
-
-    @forecasts = []
+    w_api = Wunderground.new(ENV["WUNDERGROUND_API_KEY"])
 
     # user forecast is an array of forecasts
-    current_user.selections.each do |selection|
-      station = selection.station
-      forecast = {selection_id: selection.id}
-      forecast['description'] = station.description
-
-      raw_response = w_api.forecast_and_conditions_for("pws:#{station.code}")
+    selections.each do |selection|
+      raw_response = w_api.forecast_and_conditions_for("pws:#{selection.code}")
       response = JSON.parse(raw_response.to_json, object_class: OpenStruct)
 
-      forecast['observations'] = response.current_observation
-      # convert from metric to imperial, round up, drop any float
-      # forecast['elevation'] = (forecast['observations'].display_location.elevation.to_i * 3.28084).round
-      forecast['elevation'] = forecast['observations'].observation_location.elevation
-
-
+      forecast = {
+        selection_id: selection.id,
+        description: selection.code,
+        observations: response.current_observation,
+        elevation: response.current_observation.observation_location.elevation,
+        daily_fx: {
```

Comment:
The structure of `daily_fx` could be simpler. This is what I was trying to do in the coffee shop. 

```
      forecast = {
        selection_id: selection.id,
        description: selection.code,
        observations: response.current_observation,
        elevation: response.current_observation.observation_location.elevation,
      }

      daily_forecasts = response.forecast.simpleforecast.forecastday
      forecast[:daily_fx] = daily_forecasts.map do |fx|
        {
          day: fx.date.weekday_short
          pop: fx.pop
          qpf: fx.qpf_allday.in
          snow_in: fx.snow_allday.in
          max_wind_mph: fx.maxwind.mph
        }
      end
```

Then in the view

```
      <% fcast[:daily_fx].each do |daily_fx| %>
        <tr>
          <th><%= daily_fx[:day] %></th>
          <td><%= daily_fx[:pop] %></td>
          <td><%= daily_fx[:qpf] %></td>
          <td><%= daily_fx[:snow_in]  %></td>
          <td><%= daily_fx[:max_wind_mph]  %></td>
        </tr>
      <% end %>
```

This seems better to me because there's less code and less to think about. Each element of `daily_fx` is self-contained and you don't have to keep track of your 1...3 iterator to get the right data.